homeassistant/components/wled/switch.py

```diff
@@ -0,0 +1,174 @@
+"""Support for WLED switches."""
+import logging
+from typing import Any, Callable, List
+
+from wled import WLED, WLEDError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.entity import Entity, ToggleEntity
+from homeassistant.helpers.typing import HomeAssistantType
+
+from . import WLEDDeviceEntity
+from .const import (
+    ATTR_DURATION,
+    ATTR_FADE,
+    ATTR_TARGET_BRIGHTNESS,
+    ATTR_UDP_PORT,
+    DATA_WLED_CLIENT,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+PARALLEL_UPDATES = 1
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType,
+    entry: ConfigEntry,
+    async_add_entities: Callable[[List[Entity], bool], None],
+) -> None:
+    """Set up WLED switch based on a config entry."""
+    wled: WLED = hass.data[DOMAIN][entry.entry_id][DATA_WLED_CLIENT]
+
+    switches = [
+        WLEDNightlightSwitch(entry.entry_id, wled),
+        WLEDSyncSendSwitch(entry.entry_id, wled),
+        WLEDSyncReceiveSwitch(entry.entry_id, wled),
+    ]
+    async_add_entities(switches, True)
+
+
+class WLEDSwitch(ToggleEntity, WLEDDeviceEntity):
+    """Defines a WLED switch."""
+
+    def __init__(
+        self, entry_id: str, wled: WLED, name: str, icon: str, key: str
+    ) -> None:
+        """Initialize WLED switch."""
+        self._key = key
+        self._state = False
+        super().__init__(entry_id, wled, name, icon)
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this sensor."""
+        return f"{self.wled.device.info.mac_address}_{self._key}"
+
+    @property
+    def is_on(self) -> bool:
+        """Return the state of the switch."""
+        return self._state
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn off the switch."""
+        try:
+            await self._wled_turn_off()
+            self._state = False
+        except WLEDError:
+            _LOGGER.error("An error occurred while turning off WLED switch.")
```

Comment:
We don't end logging messages with period.

Reply:
Done.